platform/measurement_utils.cpp

```diff
@@ -245,6 +245,40 @@ string FormatSpeedUnits(Units units)
   UNREACHABLE();
 }
 
+// Interleaves the bits of two 32-bit numbers.
+// The result is known as a Morton code.
+long interleave_bits(long x, long y)
+{
+  long c = 0;
+  for(int i=31; i>=0; i--)
+  {
+    c = (c << 1) | ((x >> i) & 1);
+    c = (c << 1) | ((y >> i) & 1);
+  }
+
+  return c;
+}
+
+string FormatOsmLink(double lat, double lon, int zoom)
+{
+  long x = round((lon + 180.0) / 360.0 * (1L<<32));
+  long y = round((lat + 90.0) / 180.0 * (1L<<32));
+  long code = interleave_bits(x, y);
+  string encodedCoords = "";
+  char char_array[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_~";
+  int i;
+
+  for (i = 0; i < ceil((zoom + 8) / 3.0); ++i) {
```

Comment:
- why do we use ceil and float division here? I'm not familiar with OSM links, but suspect that we can do simply (zoom + 8 + 2) / 3:
zoom = 1, res = 3
zoom = 2, res = 4
zoom = 3, res = 4
zoom = 4, res = 4
zoom = 5, res = 5

- braces { from the new line
- int i can be enclosed into loop for (int i = 0; ...)

Reply:
Original code I picked from [short_link.rb](https://github.com/openstreetmap/openstreetmap-website/blob/master/lib/short_link.rb#L57). Fixed it to use `bits::BitwiseMerge` and removed call to `ceil()`.

Reply:
No, simple append is ok here.